src/main/scala/is/hail/methods/LDPrune.scala

```diff
@@ -0,0 +1,391 @@
+        (Array((i, gpi.index)), pruneF)
+      })
+
+    val annotRDD = prunedRDD.mapValues(_ => Annotation(true)).toOrderedRDD.persist(StorageLevel.MEMORY_AND_DISK)
+    val nVariantsKept = annotRDD.count()
+
+    pruneIntermediates.foreach { gpi => gpi.rdd.unpersist() }
+    inputRDD.unpersist()
+
+    (annotRDD, nVariantsKept)
+  }
+
+  def estimateMemoryRequirements(nVariants: Long, nSamples: Int, nCores: Int, memoryPerCore: Long) = {
+    val nBytesPerVariant = math.ceil(8 * nSamples.toDouble / genotypesPerPack).toLong + variantByteOverhead
+    val memoryAvailPerCore = memoryPerCore * fractionMemoryToUse
+
+    val maxQueueSize = math.max(1, math.ceil(memoryAvailPerCore / nBytesPerVariant).toInt)
+
+    val nPartitionsMinimum = math.max(1, math.ceil(nVariants.toDouble / maxQueueSize).toInt)
+    val nPartitionsOptimal = nCores * nPartitionsPerCore
+
+    val nPartitions =
+      if (nPartitionsOptimal < nPartitionsMinimum)
+        nPartitionsMinimum
+      else
+        nPartitionsOptimal
+
+    assert(maxQueueSize > 0 && nPartitions > 0)
+
+    (maxQueueSize, nPartitions)
+  }
+
+  def apply(vds: VariantDataset, r2Threshold: Double, window: Int, nCores: Int, memoryPerCore: Long = 268435456): VariantDataset = {
+    if (nCores <= 0)
+      fatal(s"Number of cores must be greater than 0.")
+
+    if (r2Threshold < 0 || r2Threshold > 1)
+      fatal(s"R^2 threshold must be in the range [0,1]. Found `$r2Threshold'.")
+
+    if (window < 0)
+      fatal(s"Window must be greater than or equal to 0. Found `$window'.")
+
+    val sc = vds.sparkContext
+
+    val nVariantsInitial = vds.countVariants()
+    val nPartitionsInitial = vds.nPartitions
+    val nSamples = vds.nSamples
+
+    val minMemoryPerCore = math.ceil((1d / fractionMemoryToUse) * 8 * nSamples + variantByteOverhead)
+    val (maxQueueSize, _) = estimateMemoryRequirements(nVariantsInitial, nSamples, nCores, memoryPerCore)
+
+    info(s"InputData: nSamples=$nSamples nVariants=$nVariantsInitial nPartitions=$nPartitionsInitial maxQueueSize=$maxQueueSize")
+
+    if (memoryPerCore < minMemoryPerCore)
+      fatal(s"Memory per core must be greater than ${ minMemoryPerCore / (1024 * 1024) }MB")
+
+    val standardizedRDD: OrderedRDD[Locus, Variant, BitPackedVector] = vds.filterVariants { case (v, va, gs) => v.isBiallelic }.rdd
+      .flatMapValues { case (va, gs) => toBitPackedVector(gs.hardCallIterator, nSamples) }.asOrderedRDD
+
+    val ((rddLP1, nVariantsLP1, nPartitionsLP1), durationLP1) = time({
+      val prunedRDD = pruneLocal(standardizedRDD, r2Threshold, window, Option(maxQueueSize)).persist(StorageLevel.MEMORY_AND_DISK)
+      val nVariantsKept = prunedRDD.count()
+      val nPartitions = prunedRDD.partitions.length
+      assert(nVariantsKept >= 1)
+      standardizedRDD.unpersist()
+      (prunedRDD, nVariantsKept, nPartitions)
+    })
+    info(s"Local Prune 1: nVariantsKept=$nVariantsLP1 nPartitions=$nPartitionsLP1 time=${ formatTime(durationLP1) }")
```

Comment:
Again, this output may be confusing. Consider elaborating on notion of Local Prune 1, Local Prune 2, and Global Prune in docs.